### DISCUSSION TITLE
Is plugin order relevant

### DISCUSSION THREAD

**Discussion by TKDev7** [csstools/postcss-plugins#1385] [Q&A] [answered]
Does it matter how I order my plugins?

**Comment by romainmenke** [comment, marked as answer]
Plugin order has noticeable effects.
So it can matter.

Is there a specific issue you encountered?

**Reply by TKDev7** [reply]
No issues as of now. I was just wondering if I should be careful about the order.

**Reply by romainmenke** [reply]
I mostly use this order:

- bundler
- non-standard / custom things things 
- polyfills / transpilers (postcss-preset-env)
- minifier

Which you could translate as :
- combine everything
- make sure everything is standard CSS
- make sure most things work in all browsers
- make it smaller

With PostCSS 8 plugins use a visitor API that can revisit changed nodes.
So it is rare to have plugins causing conflicts.

But an optimal order can finish faster and result in a smaller output.

**Reply by TKDev7** [reply]
Much thanks! I will keep this in mind.
